contracts/contribute/Contributors.sol

```diff
@@ -38,42 +89,123 @@ struct ServiceInfo {
 /// @author Andrey Lebedev - <andrey.lebedev@valory.xyz>
 /// @author Tatiana Priemova - <tatiana.priemova@valory.xyz>
 /// @author David Vilela - <david.vilelafreire@valory.xyz>
-contract Contributors {
+contract Contributors is ERC721TokenReceiver {
```

Comment:
Makes sense to just merge these two, as the manager had a control over `Contributors` state anyway. No need to split them if they both need to be owned.